pkg/endpoint/events.go

```diff
@@ -119,6 +120,84 @@ func (e *Endpoint) PolicyRevisionBumpEvent(rev uint64) {
 	}
 }
 
+/// EndpointNoTrackEvent contains all fields necessary to update the NOTRACK rules.
+type EndpointNoTrackEvent struct {
+	ep     *Endpoint
+	annoCB AnnotationsResolverCB
+}
+
+// Handle handles the NOTRACK rule update.
+func (ev *EndpointNoTrackEvent) Handle(res chan interface{}) {
+	var oldPort uint16
+
+	e := ev.ep
+
+	// If this endpoint is going away, remove all related NOTRACK rules.
+	if err := e.lockAlive(); err != nil {
+		res <- &EndpointRegenerationResult{
+			err: nil,
+		}
+		return
+	}
+
+	defer e.unlock()
+
+	port, err := ev.annoCB(e.K8sNamespace, e.K8sPodName)
+	if err != nil {
+		res <- &EndpointRegenerationResult{
+			err: err,
+		}
+		return
+	}
+
+	// Validate annotation before we do any actual alteration to the endpoint.
+	p64, err := strconv.ParseUint(port, 10, 16)
+	// Port should be within [1-65535].
+	if err != nil || p64 == 0 || p64 > 65535 {
+		res <- &EndpointRegenerationResult{
+			err: err,
+		}
+		return
+	}
+	p := uint16(p64)
+
+	if e.noTrackPort != "" {
+		// Stored port must have good value.
+		oldPort64, _ := strconv.ParseUint(e.noTrackPort, 10, 16)
+		oldPort = uint16(oldPort64)
+	}
+
+	if port != e.noTrackPort {
+		log.Debug("Updating NOTRACK rules")
+		if e.IPv4.IsSet() {
+			if port != "" {
+				err = iptables.InstallNoTrackRules(e.IPv4.String(), p, false)
+				log.Warnf("Error installing iptable NOTRACK rules %s", err)
+			}
+			if e.noTrackPort != "" {
+				err = iptables.RemoveNoTrackRules(e.IPv4.String(), oldPort, false)
+				log.Warnf("Error removing iptable NOTRACK rules %s", err)
+			}
+		}
+		if e.IPv6.IsSet() {
+			if port != "" {
+				iptables.InstallNoTrackRules(e.IPv6.String(), p, true)
+				log.Warnf("Error installing iptable NOTRACK rules %s", err)
+			}
+			if e.noTrackPort != "" {
+				err = iptables.RemoveNoTrackRules(e.IPv6.String(), oldPort, true)
+				log.Warnf("Error removing iptable NOTRACK rules %s", err)
+			}
+		}
+		e.noTrackPort = port
```

Comment:
Any particular reason not to store `p` here rather than the string `port`? It seems like the lines 164-168 will unnecessarily re-parse the stored value later on, yet the code here seems to make use of the integer version `p`.

Reply:
Can't recall the argument that store the string value at first place, changed to store `uint16` value.